src/core/tsi/alts_transport_security.cc

```diff
@@ -24,53 +24,52 @@
 
 static alts_shared_resource g_alts_resource;
 
-static void grpc_tsi_alts_wait_for_cq_drain() {
-  while (!g_alts_resource.is_cq_drained) {
+static void wait_for_resource_destroy() {
+  gpr_mu_lock(&g_alts_resource.mu);
```

Comment:
Don't we need to lock and unlock inside of the loop below?  Otherwise, if we're holding the lock when `g_alts_resource.is_resource_destroyed` is false, `grpc_tsi_alts_shared_resource_destroy()` will never be able to get the lock, and therefore it will never be able to set `g_alts_resource.is_resource_destroyed` to true, so we'll never exit this loop.

Also, it seems odd to use both a lock and a cv for shutdown.  Isn't that redundant?

Reply:
Good point. I think the lock is not needed.